runtime/runtime.go

```diff
@@ -3246,13 +3246,19 @@ func verifyBLSPOP(
 		return false, err
 	}
 
+	// if public key is invalid, verify returns false
+	if !publicKey.IsValid {
+		return interpreter.BoolValue(false), nil
+	}
```

Comment:
The implementation of `BLSVerifyPOP` should instead check `Validated`, and validate the public key if necessary. Then, if the key is invalid, return false. 

```suggestion
```

See a detailed explanation in https://axiomzen.slack.com/archives/CG0B7CJAJ/p1637366567211300?thread_ts=1637289659.195200&cid=CG0B7CJAJ

Reply:
I think the key is guaranteed to be validated already (instantiating a PublicKey in Cadence includes the validation), so it should be safe to just check `publicKey.IsValid`.